src/PWRet.cpp

```diff
@@ -69,14 +87,31 @@ void PWRetManager::recordRuleMatch(const std::unique_ptr<IPWRuleProcessor>& proc
         key_path.PushBack(jsonKey, allocator);
     }
     param.AddMember("key_path", key_path, allocator);
-    param.AddMember("value", gather.resolvedValue, allocator);
+
     rapidjson::Value highlight, matchedValue;
     highlight.SetArray();
-    if (!gather.matchedValue.empty())
-    {
-        matchedValue.SetString(gather.matchedValue, allocator);
-        highlight.PushBack(matchedValue, allocator);
+
+    redact = redact ||
+             event_obfuscator.obfuscate_value(gather.resolvedValue) ||
+             event_obfuscator.obfuscate_value(gather.matchedValue);
+
+    if (redact) {
+        param.AddMember("value", redaction_msg, allocator);
+        if (!gather.matchedValue.empty())
+        {
+            matchedValue.SetString(redaction_msg, allocator);
+            highlight.PushBack(matchedValue, allocator);
+        }
```

Comment:
This is right only if the key_path is sensitive. Otherwise, you're supposed to only obfuscate the substrings matching the value regexp.

Pseudo example:
`password: anything` becomes `password: <redacted by datadog>`
while `key: i am a SENSITIVE value` would become `key: i am a <redacted by datadog> value` where `key` is not considered sensitive by the key regexp and the value regexp matches `SENSITIVE`

Reply:
You're right, I should've mentioned in the description that this was not implemented as the RFC. The main reason is that (as mentioned in the description) I plan to iterate on this to introduce two improvements:

- An intermediate event type, instead of generating JSON on matches.
- Obfuscation of the entire event, only when needed.

At this stage the obfuscator will actually obfuscate rather than provide a verdict and I should be able to be compliant with this.

Also, as we discussed, we won't be using value obfuscation yet so I introduced this opportunistically to avoid changing the config structure later.

Reply:
If you have no objections to this, I can open an issue so that it's clear that it'll change in the near future.

Reply:
SGTM as a first step, especially because we know the value regexp wasn't required so far for the incidents we had 👍
So feel free to remove the value regexp for now if you want to, as part of this first iteration.

Reply:
I'll keep it, but I'll make sure people are aware not to implement it for now.